Sources/BraveWallet/Crypto/Stores/TransactionDetailsStore.swift

```diff
@@ -35,30 +31,36 @@ class TransactionDetailsStore: ObservableObject, WalletObserverStore {
   private let assetRatioService: BraveWalletAssetRatioService
   private let blockchainRegistry: BraveWalletBlockchainRegistry
   private let solanaTxManagerProxy: BraveWalletSolanaTxManagerProxy
+  private let ipfsApi: IpfsAPI
   private let assetManager: WalletUserAssetManagerType
   /// Cache for storing `BlockchainToken`s that are not in user assets or our token registry.
   /// This could occur with a dapp creating a transaction.
   private var tokenInfoCache: [String: BraveWallet.BlockchainToken] = [:]
+  private var nftMetadataCache: [String: NFTMetadata] = [:]
   
   var isObserving: Bool = false
   
   init(
     transaction: BraveWallet.TransactionInfo,
+    parsedTransaction: ParsedTransaction?,
     keyringService: BraveWalletKeyringService,
     walletService: BraveWalletBraveWalletService,
     rpcService: BraveWalletJsonRpcService,
     assetRatioService: BraveWalletAssetRatioService,
     blockchainRegistry: BraveWalletBlockchainRegistry,
     solanaTxManagerProxy: BraveWalletSolanaTxManagerProxy,
+    ipfsApi: IpfsAPI,
     userAssetManager: WalletUserAssetManagerType
```

Comment:
i think we also need to add a `TxService` to listen to the change of the status in case user stays in transaction details screen for very long time.